### DISCUSSION TITLE
Error Codes

### DISCUSSION THREAD

**Discussion by rwjework** [e107inc/e107#5167] [General]
Can someone point me to a list of e107 error codes? I am getting a Error Code (22) when trying to do a mailout. Thanks

**Comment by Vodhin** [comment]
as far as I know e107 does not have a list of error codes (someone will correct me if I am wrong). Error codes are generated by the server operating system of your webhost, on which e107 runs. Error codes and their meanings vary based on what part of the OS triggered the error. e107 has a limited set of Language Files that are used by some of the e107 files to interpret the more common errors that occur.

I did a quick search for `+email+error+22` and got a lot of results that refer more to client software (like outlook), but there was one that might be the culprit and seemed to refer to secured vs unsecured connections. There are several areas where your email settings might be incorrect:

- Make sure your webhost plan allows sending email through their server via PHP, and if your host allows multiple mailouts (there may also be a limit to how many can be sent within a set time limit).

- Make sure your e107 `Preferences > Email & Contact` Info is configured to match your webhost's email capabilities (click Send Test Email To button to test)

> 1. - **Email address for emails from site**: (a valid email address on the same webhost account as your e107 installation*)
> 2. - **Emailing method**: PHP (default)
> 3. - **Default email format**:  (try Text Only)

* You can use another email host (change method from PHP to SMPT), but you may need to configure that server to allow connections from your e107 server as well.
Hope that helps...

**Comment by Deltik** [comment]
The string "Error Code" (case-sensitive) does not appear in the e107 codebase.  Can you share what page you were on and the exact error message you encountered?

Otherwise, we can only speculate about what error codes you mean.
